### PR TITLE
Fixes for scippnexus

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Fixes for 0.4 (will need 0.4.1, see https://github.com/scipp/scippnexus/pull/76).

### CODE REVIEW

File: src/scippneutron/file_loading/load_nexus.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def _match_nx_class(_, node):
         names = [group.name.split('/')[-1] for group in groups]
         if len(names) != len(set(names)):  # fall back to full path if duplicate
             names = [group.name for group in groups]
-        out[nx_class] = {n: _make(g) for n, g in zip(names, groups)}
+        out[nx_class] = {n: group._make(g) for n, g in zip(names, groups)}
```
Comment (jl-wynen) [inline src/scippneutron/file_loading/load_nexus.py:189]: Is it really necessary to use private members? (`_make_`, `nx_class_registry`)

Comment (SimonHeybrock) [reply src/scippneutron/file_loading/load_nexus.py:189]: Depends on whether `load_nexus` will survive. If we plan to support it long term then we should indeed to move to a public API.

Comment (jl-wynen) [reply src/scippneutron/file_loading/load_nexus.py:189]: Ok. that is a discussion for another time.

**Comment by SimonHeybrock** [comment]
Builds seem to be working now. Do we want to make a release after merging?

**Review by jl-wynen — APPROVED** [review]

**Comment by jl-wynen** [comment]
> Do we want to make a release after merging?

Sure

**Pull request merged**